### ISSUE TITLE
[Core] ray2.3.1 gcs_server memory keeps increasing until OOM

### ISSUE THREAD

**Issue by MissiontoMars** [ray-project/ray#34619]
### What happened + What you expected to happen

The way to reproduce the issue is described here: https://discuss.ray.io/t/how-to-get-gcs-server-momery-distribution-to-debug-memory-continued-increasement/10030

### Versions / Dependencies

pip install “ray[default]”==2.3.1
without any code change.

### Reproduction script

None

### Issue Severity

High: It blocks me from completing my task.

**Comment by MissiontoMars** [comment]
@iycheng @scv119 Could you help me address this issue?

**Comment by rkooo567** [comment]
Since it is not a big memory leak (15MB), and we stores some metadata that we don't clean up per each job run, we will downgrade the priority to P1. But @iycheng will look into it.

**Comment by MissiontoMars** [comment]
> 

Does GCS have a elimination mechanism for job metadata now? We treat ray cluster as long running cluster, lots of job will be submitted to the cluster. GCS oom is unacceptable.

**Comment by MissiontoMars** [comment]
> Since it is not a big memory leak (15MB), and we stores some metadata that we don't clean up per each job run, we will downgrade the priority to P1. But @iycheng will look into it.

Does GCS have a elimination mechanism for job metadata? We treat the ray cluster as a long-running cluster, lots of job will be submitted to he cluster. So gcs oom seems unacceptable.

**Comment by rkooo567** [comment]
Yes. It removes actor and task metadata eventually. We only caches up to 10K entries for actors, and 100K entries for tasks.

I believe there are other data related to the job that's GC'ed as a whole, not fate shared with a job. It'd be great if you have a chance to memory profiler the GCS after the job termination and let us know what memory was leaked (at least I believe this is what Yi will do).

**Comment by cszhu** [comment]
This P2 issue has seen no activity in the past 2 years. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please comment and remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
